Analysis/Tasks/PWGHF/taskXic.cxx

```diff
@@ -0,0 +1,171 @@
+        //Printf("Candidate: Y rejection: %g", YXic(candidate));
+        continue;
+      }
+      if (candidate.isSelXicToPKPi() >= d_selectionFlagXic) {
+        registry.fill(HIST("hmass"), InvMassXicToPKPi(candidate));
+      }
+      if (candidate.isSelXicToPiKP() >= d_selectionFlagXic) {
+        registry.fill(HIST("hmass"), InvMassXicToPiKP(candidate));
+      }
+      registry.fill(HIST("hptcand"), candidate.pt());
+      registry.fill(HIST("hptprong0"), candidate.ptProng0());
+      registry.fill(HIST("hptprong1"), candidate.ptProng1());
+      registry.fill(HIST("hptprong2"), candidate.ptProng2());
+      registry.fill(HIST("hdeclength"), candidate.decayLength());
+      registry.fill(HIST("hd0Prong0"), candidate.impactParameter0());
+      registry.fill(HIST("hd0Prong1"), candidate.impactParameter1());
+      registry.fill(HIST("hd0Prong2"), candidate.impactParameter2());
+      registry.fill(HIST("hCt"), CtXic(candidate));
+      registry.fill(HIST("hCPA"), candidate.cpa());
+      registry.fill(HIST("hEta"), candidate.eta());
+      registry.fill(HIST("hselectionstatus"), candidate.isSelXicToPKPi());
+      registry.fill(HIST("hselectionstatus"), candidate.isSelXicToPiKP());
+      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter0());
+      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter1());
+      registry.fill(HIST("hImpParErr"), candidate.errorImpactParameter2());
+      registry.fill(HIST("hDecLenErr"), candidate.errorDecayLength());
+      registry.fill(HIST("hDecLenErr"), candidate.chi2PCA());
+    }
+  }
+};
+
+/// Fills MC histograms.
+struct TaskXicMC {
+  HistogramRegistry registry{
+    "registry",
+    {{"hPtRecSig", "3-prong candidates (matched);#it{p}_{T}^{rec.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtRecBg", "3-prong candidates (unmatched);#it{p}_{T}^{rec.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtGen", "MC particles (matched);#it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtGenSig", "3-prong candidates (matched);#it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hCPARecSig", "3-prong candidates (matched);cosine of pointing angle;entries", {HistType::kTH1F, {{110, -1.1, 1.1}}}},
+     {"hCPARecBg", "3-prong candidates (unmatched);cosine of pointing angle;entries", {HistType::kTH1F, {{110, -1.1, 1.1}}}},
+     {"hEtaRecSig", "3-prong candidates (matched);#it{#eta};entries", {HistType::kTH1F, {{100, -2., 2.}}}},
+     {"hEtaRecBg", "3-prong candidates (unmatched);#it{#eta};entries", {HistType::kTH1F, {{100, -2., 2.}}}},
+     {"hEtaGen", "MC particles (matched);#it{#eta};entries", {HistType::kTH1F, {{100, -2., 2.}}}}}};
+
+  Configurable<int> d_selectionFlagXic{"d_selectionFlagXic", 1, "Selection Flag for Xic"};
+  Configurable<int> d_selectionFlagXicbar{"d_selectionFlagXicbar", 1, "Selection Flag for Xicbar"};
```

Comment:
This d_selectionFlagXicbar is obsolete i think, probably a leftover. I would remove it 
(same for Lc) @mfaggin @DelloStritto

Reply:
@ginnocen I removed that line